core/src/main/scala/io/finch/Endpoint.scala

```diff
@@ -478,6 +478,7 @@ object Endpoint {
    * The resulting endpoint will fail when type conversion fails.
    */
   implicit class StringEndpointOps(val self: Endpoint[String]) extends AnyVal {
+    @deprecated(s"Use type parameter instead for a corresponding endpoint (param[A], header[A], ...)", "0.16")
```

Comment:
We should also deprecate `as` under `StringOptionEndpointOps`, `StringSeqEndpointOps`, and `StringNelEndpointOps`.

Reply:
What about `ValueEndpointOps`?

I've made a generic `DecodeEntity` for this case:
https://github.com/ImLiar/finch/blob/47d05844f323ac1bc5cfda1208ba302f6540ccfe/core/src/main/scala/io/finch/DecodeEntity.scala#L56

So it's possible to use `param[Foo]` now.

Reply:
Yeah, let's keep it. I think we still want to support `param[String].as[Foo]` yet I'm not sure we have to support `param[Foo]` (unless there is `DecodeEntity[Foo]` available).

Reply:
This implicit does exactly the same. It lifts parameter into a single value case class `Foo` for free.

Reply:
Yeah I know. There was a comment on that instances saying it's a workaround for `param[String].as[Foo]`, which otherwise doesn't work (even though there is `ValueEndpointOps` that's basically a noop right now).